### PR TITLE
Bismillah module submissions

### PR DESCRIPTION
**Pull request by smellslikekeenspirit**

### CODE REVIEW

File: server/src/db/migrations/20211102194503_machines_table.ts [hunk 1]
```diff
@@ -26,15 +26,30 @@ export async function up(knex: Knex): Promise<void> {
...
       if (!exists) {
-        return knex.schema.createTable("Reservations", function (t) {
-          t.increments("id").primary();
-          t.integer("userId").references("id").inTable("Users");
-          t.integer("equipmentId").references("id").inTable("Equipment");
-          t.timestamp("createdAt").defaultTo(knex.fn.now());
-          t.time("startTime");
-          t.time("endTime");
+        return knex.schema.createTable('Reservations', function (t) {
+          t.increments('id');
```
Comment (pjw9963) [inline server/src/db/migrations/20211102194503_machines_table.ts:32]: add .primary() for the id fields like you did with the module submission table

File: server/src/db/migrations/20211102194503_machines_table.ts [hunk 2]
```diff
@@ -26,15 +26,30 @@ export async function up(knex: Knex): Promise<void> {
...
+          t.time('endTime');
+          t.integer('equipment').references('id').inTable('Equipment');
+          t.enu('status', ['PENDING', 'CONFIRMED', 'CANCELLED']);
+          t.timestamp('lastUpdated').defaultTo(knex.fn.now());
+        });
+      }
+    }).then(async () => {
+      const exists = await knex.schema.hasTable('ReservationEvents');
+      if (!exists) {
+        return knex.schema.createTable('ReservationEvents', function (t) {
```
Comment (pjw9963) [inline server/src/db/migrations/20211102194503_machines_table.ts:47]: I feel like there should be a foreign key on this table to reference the reservation the event is for

**Review by pjw9963 — CHANGES_REQUESTED** [review]
Little things. Let me know if I'm wrong about the FK

File: server/src/db/migrations/20211102194503_machines_table.ts (hunk 2 above)
Comment (smellslikekeenspirit) [reply server/src/db/migrations/20211102194503_machines_table.ts:47]: @pjw9963 right, I didn't mean to include this Reservation stuff, I've mixed up some of the stuff from my other PR. But fixing now that I have

File: server/src/db/migrations/20220324210801_module_submissions_table.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema
+    .hasTable("ModuleSubmissions")
+    .then(function (exists) {
+      if (!exists) {
+        return knex.schema.createTable("ModuleSubmissions", function (t) {
+          t.increments("id").primary();
+          t.integer("moduleId").references("id").inTable("TrainingModule");
+          t.integer('makerId').references('id').inTable('Users');
```
Comment (MatthewGalan) [inline server/src/db/migrations/20220324210801_module_submissions_table.ts:11]: Can we rename these to `moduleID` and `makerID`?

**Review by MatthewGalan — CHANGES_REQUESTED** [review]

File: server/src/db/migrations/20220324210801_module_submissions_table.ts (hunk 3 above)
Comment (smellslikekeenspirit) [reply server/src/db/migrations/20220324210801_module_submissions_table.ts:11]: @MatthewGalan it's `FoobarId` everywhere though, maybe we should change those too

**Comment by smellslikekeenspirit** [comment]
changed name to something more auspicious

File: server/src/db/migrations/20220324210801_module_submissions_table.ts (hunk 3 above)
Comment (MatthewGalan) [reply server/src/db/migrations/20220324210801_module_submissions_table.ts]: Let's slowly migrate those over as well. We took a poll on the issue a while ago: https://discord.com/channels/879846554540855376/879846554540855379/949763875413303366

**Review by MatthewGalan — APPROVED** [review]

**Pull request merged**